### PR TITLE
MIG-1509: MTC 1.7.15 Release Notes

### PR DESCRIPTION
**Pull request by anarnold97**
### JIRA

* [MIG-1509](https://issues.redhat.com/browse/MIG-1509)

* GA 30th April

### Version(s):

* OCP 4.12 → branch/enterprise-4.12
* OCP 4.13 → branch/enterprise-4.13
* OCP 4.14 → branch/enterprise-4.14
* OCP 4.15 → branch/enterprise-4.15
* OCP 4.16 → branch/enterprise-4.16

### Link to docs preview:

* [MTC 1.7.15 release notes](https://75137--ocpdocs-pr.netlify.app/openshift-enterprise/latest/migration_toolkit_for_containers/mtc-release-notes.html#migration-mtc-release-notes-1-7-15_mtc-release-notes)

QE review:
- [X ] [QE has approved this change.](https://github.com/openshift/openshift-docs/pull/75137#pullrequestreview-2027504824)

Additional information:

### CODE REVIEW

File: modules/migration-mtc-release-notes-1-7-15.adoc [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+To resolve this issue, upgrade to {mtc-short} 1.7.15.
+
+For more details, see link:https://access.redhat.com/security/cve/CVE-2024-28180[(CVE-2024-28180)]
+
+
+
+[id="known-issues-1-7-15_{context}"]
+== Known issues
+
+There are no major known issues in this release.
```
Comment (anarnold97) [inline modules/migration-mtc-release-notes-1-7-15.adoc:35]:
```suggestion
This release has the following known issues:
```

**Review by stillalearner — APPROVED** [review]
lgtm

**Comment by dfitzmau** [comment]
Hi @anarnold97 . Please tick the QE box if they have approved this PR?

File: modules/migration-mtc-release-notes-1-7-15.adoc [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+= {mtc-full} 1.7.15 release notes
+
+[id="resolved-issues-1-7-15_{context}"]
+== Resolved issues
+
+This release has the following resolved issues:
+
+.CVE-2024-24786: A flaw was found in Golang's protobuf module, where the unmarshal function can enter an infinite loop
+
+A flaw was found in the `protojson.Unmarshal` function, which could enter an infinite loop when unmarshaling certain forms of invalid JSON. This condition could occur when unmarshaling into a message that contained a `google.protobuf.Any` value or when the `UnmarshalOptions.DiscardUnknown` option was set.
```
Comment (dfitzmau) [inline modules/migration-mtc-release-notes-1-7-15.adoc:15]: "which" is typically used for non-restrictive clauses but this statement seems essential· How about:

> A flaw was found in the `protojson.Unmarshal` function that could cause the function to enter an infinite loop when unmarshaling certain forms of invalid JSON messages. This condition could occur when unmarshaling into a message that contained a `google.protobuf.Any` value or when the `UnmarshalOptions.DiscardUnknown` option was set in a JSON-formatted message.

File: modules/migration-mtc-release-notes-1-7-15.adoc [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+To resolve this issue, upgrade to {mtc-short} 1.7.15.
+
+For more details, see link:https://access.redhat.com/security/cve/CVE-2024-24786[(CVE-2024-24786)].
+
+.CVE-2024-28180: `jose-go` improper handling of highly compressed data
+
+A vulnerability was found in Jose due to improper handling of highly compressed data. An attacker could send a JSON Web Encryption (JWE) containing compressed data that used large amounts of memory and CPU when decompressed by Decrypt or DecryptMulti.
```
Comment (dfitzmau) [inline modules/migration-mtc-release-notes-1-7-15.adoc:23]:
```suggestion
A vulnerability was found in Jose due to improper handling of highly compressed data. An attacker could send a JSON Web Encryption (JWE) encrypted message that contained compressed data that used large amounts of memory and CPU when decompressed by `Decrypt` or `DecryptMulti` vector/function. 
```

Comment (dfitzmau) [reply modules/migration-mtc-release-notes-1-7-15.adoc:23]: Vector or function?

File: modules/migration-mtc-release-notes-1-7-15.adoc [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+
+[id="known-issues-1-7-15_{context}"]
+== Known issues
+
+This release has the following known issues:
+
+.Direct Volume Migration is failing as the rsync pod on the source cluster goes into an `Error` state
+
+On migrating any application with Persistent Volume Claim (PVC), Stage succeeds with warnings and Direct Volume Migration (DVM) fails with the `rsync` pod on the source namespace going into an `error` state. link:https://bugzilla.redhat.com/show_bug.cgi?id=2256141[(BZ#2256141)]
```
Comment (dfitzmau) [inline modules/migration-mtc-release-notes-1-7-15.adoc:37]: "On migrating any application with Persistent Volume Claim (PVC)"

Not sure what is meant here:

> An application that contains a PVC or migrating an application by using a PVC.

File: modules/migration-mtc-release-notes-1-7-15.adoc [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+For more details, see link:https://access.redhat.com/security/cve/CVE-2024-28180[(CVE-2024-28180)].
+
+
+[id="known-issues-1-7-15_{context}"]
+== Known issues
+
+This release has the following known issues:
+
+.Direct Volume Migration is failing as the rsync pod on the source cluster goes into an `Error` state
```
Comment (dfitzmau) [inline modules/migration-mtc-release-notes-1-7-15.adoc:35]:
```suggestion
.Direct Volume Migration is failing as the Rsync pod on the source cluster goes into an `Error` state
```

Comment (dfitzmau) [reply modules/migration-mtc-release-notes-1-7-15.adoc:35]: As per https://docs.openshift.com/container-platform/4.15/migration_toolkit_for_containers/advanced-migration-options-mtc.html ?

File: modules/migration-mtc-release-notes-1-7-15.adoc (hunk 4 above)
Comment (dfitzmau) [reply modules/migration-mtc-release-notes-1-7-15.adoc:37]: "Stage" is a migration operation? Might be worth stating this so a supporting noun is evident.

File: modules/migration-mtc-release-notes-1-7-15.adoc [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+On migrating any application with Persistent Volume Claim (PVC), Stage succeeds with warnings and Direct Volume Migration (DVM) fails with the `rsync` pod on the source namespace going into an `error` state. link:https://bugzilla.redhat.com/show_bug.cgi?id=2256141[(BZ#2256141)]
+
+.The conflict condition is briefly cleared after it is created
+
+When creating a new state migration plan that will result in a conflict error, that error is cleared shortly after it is displayed. link:https://bugzilla.redhat.com/show_bug.cgi?id=2144299[(BZ#2144299)]
```
Comment (dfitzmau) [inline modules/migration-mtc-release-notes-1-7-15.adoc:41]:
```suggestion
When creating a new state migration plan that results in a conflict error message, the error message is cleared shortly after it is displayed. link:https://bugzilla.redhat.com/show_bug.cgi?id=2144299[(BZ#2144299)]
```

Displayed on a terminal?

File: modules/migration-mtc-release-notes-1-7-15.adoc [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+.Migration fails when there are multiple Volume Snapshot Locations (VSLs) of different provider types configured in a cluster and no default VSL is set
```
Comment (dfitzmau) [inline modules/migration-mtc-release-notes-1-7-15.adoc:43]:
```suggestion
.Migration fails when there are multiple Volume Snapshot Locations (VSLs) of different provider types configured in a cluster with no specified default VSL.
```

File: modules/migration-mtc-release-notes-1-7-15.adoc [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+When there are multiple VSLs in a cluster with different provider types, and none of them is set as default, Velero results in a validation error, making migrations fail.  link:https://bugzilla.redhat.com/show_bug.cgi?id=2180565[(BZ#2180565)]
```
Comment (dfitzmau) [inline modules/migration-mtc-release-notes-1-7-15.adoc:45]:
```suggestion
When there are multiple VSLs in a cluster with different provider types, and you set none of them  as the default VSL, Velero results in a validation error that cuases migration operations to fail.  link:https://bugzilla.redhat.com/show_bug.cgi?id=2180565[(BZ#2180565)]
```

**Comment by dfitzmau** [comment]
Hi @anarnold97 . Great work. I added minor comments inline.

/remove-label peer-review-in-progress

/label peer-review-done

File: modules/migration-mtc-release-notes-1-7-15.adoc (hunk 2 above)
Comment (anarnold97) [inline modules/migration-mtc-release-notes-1-7-15.adoc:15]:
```suggestion
A flaw was found in the `protojson.Unmarshal` function that could cause the function to enter an infinite loop when unmarshaling certain forms of invalid JSON messages. This condition could occur when unmarshaling into a message that contained a `google.protobuf.Any` value or when the `UnmarshalOptions.DiscardUnknown` option was set in a JSON-formatted message.
```

File: modules/migration-mtc-release-notes-1-7-15.adoc [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+On migrating any application with Persistent Volume Claim (PVC), Stage succeeds with warnings and Direct Volume Migration (DVM) fails with the `rsync` pod on the source namespace going into an `error` state. link:https://bugzilla.redhat.com/show_bug.cgi?id=2256141[(BZ#2256141)]
+
+.The conflict condition is briefly cleared after it is created
+
+When creating a new state migration plan that results in a conflict error message, the error message is cleared shortly after it is displayed. link:https://bugzilla.redhat.com/show_bug.cgi?id=2144299[(BZ#2144299)]
+
+.Migration fails when there are multiple Volume Snapshot Locations (VSLs) of different provider types configured in a cluster with no specified default VSL.
+
+When there are multiple VSLs in a cluster with different provider types, and none of them is set as default, Velero results in a validation error, making migrations fail.  link:https://bugzilla.redhat.com/show_bug.cgi?id=2180565[(BZ#2180565)]
```
Comment (anarnold97) [inline modules/migration-mtc-release-notes-1-7-15.adoc:45]:
```suggestion
When there are multiple VSLs in a cluster with different provider types, and you set none of them as the default VSL, Velero results in a validation error that causes migration operations to fail. link:https://bugzilla.redhat.com/show_bug.cgi?id=2180565[(BZ#2180565)]
```

**Comment by mburke5678** [comment]
Merge review looks good to me. **_DO NOT MERGE until MTC 1.7.15 goes GA._**

**Comment by anarnold97** [comment]
@mburke5678 - thanks for doing the extra

**Comment by anarnold97** [comment]
/label merge-review-needed

SHIPPED_LIVE
https://catalog.redhat.com/software/containers/rhmtc/openshift-migration-operator-bundle/5f5a277c31e418908b56afcf?architecture=amd64&image=66268047864618d041feeee8

**Pull request merged**